examples/semantic_textual_similarity/semantic_textual_similarity/evaluator.py

```diff
@@ -0,0 +1,172 @@
+from kolena.workflow.evaluator_function import TestCases
+
+
+def compute_test_sample_metrics(gt: GroundTruth, inf: Inference) -> TestSampleMetric:
+    return TestSampleMetric(
+        error=gt.similarity - inf.similarity,
+        abs_error=abs(gt.similarity - inf.similarity),
+    )
+
+
+def compute_aggregate_metrics(
+    test_samples_metrics: List[TestSampleMetric],
+    ground_truths: List[GroundTruth],
+    inferences: List[Inference],
+) -> TestCaseMetric:
+    gts = pd.Series([gt.similarity for gt in ground_truths])
+    infs = pd.Series([inf.similarity for inf in inferences])
+
+    abs_errors = np.array([metric.abs_error for metric in test_samples_metrics])
+    abs_errors_squared = np.array([metric.abs_error**2 for metric in test_samples_metrics])
+
+    return TestCaseMetric(
+        PearsonCorr=gts.corr(infs, method="pearson"),
+        SpearmanCorr=gts.corr(infs, method="spearman"),
+        MAE=abs_errors.mean(),
+        RMSE=np.sqrt(abs_errors_squared.mean()),
+    )
+
+
+def compute_score_distribution_plot(
+    score: str,
+    metrics: List[Union[SentencePair, TestSampleMetric]],
+    binning_info: Optional[Tuple[float, float, float]] = None,  # start, end, num
+) -> Histogram:
+    scores = [getattr(m, score) for m in metrics]
+    bins = np.linspace(*binning_info)
+
+    hist, _ = np.histogram(scores, bins=bins)
+    return Histogram(
+        title=f"Distribution of {score}",
+        x_label=f"{score}",
+        y_label="Count",
+        buckets=list(bins),
+        frequency=list(hist),
+    )
+
+
+def compute_metric_vs_metric_plot(
+    x_metric: str,
+    y_metric: str,
+    x_values: List[Union[int, float]],
+    y_values: List[Union[int, float]],
+    binning_info: Optional[Tuple[float, float, float]] = None,  # start, end, num
+) -> CurvePlot:
+    bins = list(np.linspace(*binning_info))
+
+    bins_centers: List[float] = []
+    bins_values: List[float] = []
+    for i in range(len(bins) - 1):
+        lo, hi = bins[i : i + 2]
+        bin_values = [y for y, x in zip(y_values, x_values) if lo <= x < hi]
+        if len(bin_values) > 0:
+            bins_centers.append(lo + ((hi - lo) / 2))
+            bins_values.append(np.mean(bin_values))
+
+    return CurvePlot(
+        title=f"{y_metric} vs. {x_metric}",
+        x_label=f"{x_metric}",
+        y_label=f"{y_metric}",
+        curves=[Curve(x=bins_centers, y=bins_values)],
+    )
+
+
+def compute_plots(
+    metrics: List[TestSampleMetric],
+    test_samples: List[SentencePair],
+) -> List[Plot]:
+    return [
+        compute_score_distribution_plot(
+            "error",
+            metrics,
+            (-1.0, 1.0, 21),
+        ),
+        compute_score_distribution_plot(
+            "word_count_diff",
+            test_samples,
+            (0, 20, 21),
+        ),
+        compute_score_distribution_plot(
+            "char_length_diff",
+            test_samples,
+            (0, 140, 71),
+        ),
```

Comment:
Are these necessary? Each is automatically present in the Studio distributions viewer.

Reply:
Hmm true. It's just nicer to have them on the same page when looking at the results but I can drop these.